src/wireframes/renderer/ItemsLayer.tsx

```diff
@@ -36,8 +36,10 @@ export const ItemsLayer = React.memo((props: ItemsLayerProps) => {
     const shapesRendered = React.useRef(onRender);
     const shapeRefsById = React.useRef<Record<string, ItemWithPreview>>({});
 
-    const itemIds = diagram?.rootIds;
-    const items = diagram?.items;
+    const nestedRootIds = diagrams.map(d=>d.rootIds);
+    const itemIds =  React.useMemo(()=> new ImmutableList<string>(nestedRootIds.flatMap(x=>x.values)), [nestedRootIds]);
```

Comment:
All thesep roperties are only needed inside the orderedShapes. Unnecessary allocations in my opinion, because a few nested loops would be easier.